olive/model/utils/hf_mappings.py

```diff
@@ -73,6 +73,7 @@
 # there is the priority order: NUM_HEADS_NAMES[0] and HIDDEN_SIZE_NAMES[0] are the first choice
 # which means user can override the value in config file
 NUM_HEADS_NAMES = ("num_heads", "num_attention_heads", "n_head", "n_heads", "encoder_attention_heads")
+NUM_HIDDEN_LAYER_NAMES = ("num_hidden_layers", "num_layers", "n_layer", "n_layers")
```

Comment:
>NUM_HIDDEN_LAYER_NAMES [](http://example.com/codeflow?start=0&length=22)

could we let each model specify the hidden/head names instead of speculate by Olive using the central place?

Reply:
in this way, Olive will not need maintain the head names/hidden layer names whenever there are new model released.

Reply:
We have that and expose the config to user in kv_cache_config if they are clear the detailed value. https://github.com/microsoft/Olive/pull/1121/files#diff-a9c02489c56cfee4594be95735d9edd61297573f46c68581ba6e38556020415bR20

Reply:
https://github.com/microsoft/Olive/pull/1121/files#r1582052278. This is what I mean, the first get value might not be correct.

Reply:
for example, we could choose the most populate name "num_hidden_layers" as default. for all other names like n_layer, user need specify the name or value by kv_cache config explicitly

Reply:
I think value is enough, if user knows the name. he must know the value.

Reply:
If user specify the value, the value will exists in both config and model attributes, which has the possibility that they are different.

Reply:
What kind of case they are different? For huggingface model attributes, it is pulled from huggingface hub. For the config, it should be same with the value in huggingface hub's config.json. 
If the values are different, I think it is user's fault? User should take the responsibility.

Reply:
as for difference, I mean config redundancy since there are multiple places to have the config